### PR TITLE
NAS-132196 / 25.04 / Fix bugs in ACME DNS auth service and convert schemas to new api style

### PR DESCRIPTION
**Pull request by sonicaj**
This PR adds following changes:

1. Refactors some code to reflect new guidelines we have in place for defining pydantic models of services
2. Fixes some bugs with existing models with key mismatches and args handling
3. Convert authenticator schemas to use new api style pydantic models

### CODE REVIEW

File: src/middlewared/middlewared/api/base/types/base/path.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from pathlib import Path
```
Comment (yocalebo) [inline src/middlewared/middlewared/api/base/types/base/path.py:1]: This entire file can be removed and replaced with the built-in `FilePath` builtin pydantic annotation. Please use that instead.

**Review by yocalebo — CHANGES_REQUESTED** [review]

**Comment by creatorcary** [comment]
Why are we adding `validate_credentials` to our versioned api? (Are they public endpoints?)

**Comment by sonicaj** [comment]
@creatorcary they are not middleware endpoints at all, it is just making use of the accepts/returns model with pure python as these are not services themselves, they will never get collected by middleware. @themylogin do you think we should have another util which `api_method` can call and that can hold this logic ?

**Comment by themylogin** [comment]
@sonicaj it’s better to have all models be accepted immediately by API methods. The reason is: if models change between API definitions, who will convert them? Please, try to fit these models into the function calls themselves using a discriminator (that’s already done for new VMs, and that’s what I’ll soon do for alerts and cloud syncs).

**Review by themylogin — APPROVED** [review]
Excellent, thank you!

**Review by yocalebo — APPROVED** [review]

**Pull request merged**